### ISSUE TITLE
Add label  from http headers through vmagent or vmauth

### ISSUE THREAD

**Issue by 131** [VictoriaMetrics/VictoriaMetrics#5719]
### Is your question request related to a specific component?

vmagent or vmauth

### Describe the question in detail

In my web cluster, I’m using an external (nginx)  client certs & ca ssl terminaison load balancer as entrypoint for my IoT devices’ vmagent’s remote write url.

In this cluster, after the nginx engine and  before reaching VictoriaMetrics backend, i want to extends/tag all metrics  through either a cloud vmagent or vmauth microservice and use extras HTTP headers, added by nginx, as labels to all ingested metrics. 
I think this is a basic feature, but could not find example/references to to meta __http_request_header in relabeling configuration
Can you guide me here please ?

### Troubleshooting docs

- [X] General - https://docs.victoriametrics.com/Troubleshooting.html
- [X] vmagent - https://docs.victoriametrics.com/vmagent.html#troubleshooting
- [X] vmalert - https://docs.victoriametrics.com/vmalert.html#troubleshooting

**Comment by Haleygo** [comment]
Hello!
>i want to extends/tag all metrics through either a cloud vmagent or vmauth microservice and use extras HTTP headers, added by nginx, as labels to all ingested metrics.

vmauth is a simple proxy which shouldn't change the content of requests, so expanding labels is out of it's scope.

Are those header labels you want to add static? If they are, you can use `external_labels` in [vmagent](https://new.docs.victoriametrics.com/vmagent/#adding-labels-to-metrics).
And vmagent won't attach http requests headers to metrics since it costs extra resources and not normally needed by users, so you can't find it in relabeling configuration.

But it shouldn't be hard to write a http proxy which read the requests header and append to series.

**Comment by Haleygo** [comment]
>And i understand -and i just tested- that the extra_label query parameters are merged as label in write api.

Yeah, you have a vmagent in front of nginx which can add extra_label, right?
You can also add extra vmagent after nginx.

**Comment by nbragin4** [comment]
Bump. Any news about this feature? @131

**Comment by nulldoot2k** [comment]
@Haleygo param extra_label can support multi values?
for example:
- "http://vmselect1:8481/select/10/prometheus?extra_label=db=serviceA,serviceB,serviceC" ??
before inside config relabels vmagent
```
- source_labels: [monitor]
  regex: (^serviceA|serviceB|serviceC|serviceD|...$)
  action: keep
```

**Comment by nulldoot2k** [comment]
Ah i fixed id :)
Alertnative extra_label, using extra_filters :)))

**Comment by nulldoot2k** [comment]
I open issue: https://github.com/VictoriaMetrics/VictoriaMetrics/issues/8552

**Comment by Haleygo** [comment]
>Using vm behind a reverse proxy is common (and documented) an example of extra metrics would have help me

As mentioned in https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5719#issuecomment-1916397831, it's not a common practice to read the http requests headers and convert them into metric labels, so it won't be supported natively by vmagent/vmauth or added to the documentation.
Close this as answered, feel free to reopen or start a new issue if there are further questions.
